plugins/woocommerce/src/Internal/Traits/AccessiblePrivateMethods.php

```
@@ -0,0 +1,210 @@
+			$this->mark_method_as_accessible( $callback[1] );
+		}
+		add_filter( $hook_name, $callback, $priority, $accepted_args );
+	}
+
+	/**
+	 * Register a WordPress filter.
+	 * If the callback refers to a private or protected static method in this class, the method is marked as externally accessible.
+	 *
+	 * $callback can be a standard callable, or a string representing the name of a method in this class.
+	 *
+	 * @param string   $hook_name       The name of the action to add the callback to.
+	 * @param callable $callback        The callback to be run when the action is called.
+	 * @param int      $priority        Optional. Used to specify the order in which the functions
+	 *                                  associated with a particular action are executed.
+	 *                                  Lower numbers correspond with earlier execution,
+	 *                                  and functions with the same priority are executed
+	 *                                  in the order in which they were added to the action. Default 10.
+	 * @param int      $accepted_args   Optional. The number of arguments the function accepts. Default 1.
+	 */
+	protected static function add_static_filter( string $hook_name, $callback, int $priority = 10, int $accepted_args = 1 ): void {
+		if ( is_string( $callback ) && method_exists( __CLASS__, $callback ) ) {
+			ArrayUtil::push_once( static::$_accessible_static_private_methods, $callback );
+			$callback = array( __CLASS__, $callback );
+		} elseif ( is_array( $callback ) && count( $callback ) >= 2 && __CLASS__ === $callback[0] ) {
+			static::mark_static_method_as_accessible( $callback[1] );
+		}
+		add_filter( $hook_name, $callback, $priority, $accepted_args );
+	}
+
+	/**
+	 * Register a private or protected instance method of this class as externally accessible.
+	 *
+	 * @param string $method_name Method name.
+	 * @return void
+	 */
+	protected function mark_method_as_accessible( string $method_name ): void {
```

I know we're trying to make our methods work in a similar way to WordPress's and for that we don't need any warnings or anything, but the fact that this just silently does nothing when the `$method` doesn't exist makes me a bit uneasy.

---

An easy solution here could be to return a success/failure boolean.